faust_large_message_serializer/serializer.py

```diff
@@ -0,0 +1,33 @@
+from faust.serializers.codecs import Codec
+
+from faust_large_message_serializer.blob_storage.blog_storage_factory import (
+    BlobStorageFactory,
+)
+from faust_large_message_serializer.clients.retrieving_client import RetrievingClient
+from faust_large_message_serializer.clients.storing_client import StoringClient
+from faust_large_message_serializer.config import LargeMessageSerializerConfig
+
+
+class LargeMessageSerializer(Codec):
+    def __init__(
+        self,
+        output_topic: str,
+        config: LargeMessageSerializerConfig,
+        is_key: bool = False,
+        **kwargs
+    ):
+        super().__init__(
+            output_topic=output_topic, config=config, is_key=is_key, **kwargs
+        )
+        self._output_topic = output_topic
+        self._config = config
+        self._is_key = is_key
+        self._blob_factory = BlobStorageFactory(config)
+        self._storage_client = StoringClient(config, self._blob_factory)
```

Comment:
Isn't it better to just have sth like config.create_storing_client()?

Reply:
I mean, I was thinkig about an only data config storage without too many logic inside. That's why I thought that inverse the dependencies and having a class that only construct the client would be a good idea.

Reply:
I disagree, the StoringClient should just receive the built BlobStorageClient and the whole logic which configs are available and used where belongs in the config object. We also build object oriented software, not only data classes :)

Reply:
ok

Reply:
@philipp94831 better ?